site/en/blog/renderingng-data-structures/index.md

```diff
@@ -0,0 +1,747 @@
+drawing each one with the specified visual effects,
+to produce the desired pixel output for the render pass.
+Compositing draw quads for a render pass can be done efficiently on the GPU,
+because the allowed visual effects are carefully chosen to be those that map directly to GPU features.
+
+There are additional types of draw quads beyond rasterized tiles.
+For example, there are solid color draw quads that are not backed by a texture at all,
+or texture draw quads for non-tile textures like video or canvas.
+
+It is also possible for a compositor frame to embed another compositor frame.
+For example, the browser compositor produces a compositor frame with the browser UI,
+and an empty rectangle where the render compositor content will be embedded.
+Another example is site isolated iframes. This embedding is accomplished through _surfaces_.
+
+When a compositor submits a compositor frame, it is accompanied by an identifier,
+called a _surface id_, allowing other compositor frames to embed it by reference.
+The newest compositor frame submitted with a particular surface id is stored by Viz.
+Another compositor frame can then refer to it later via a _surface draw quad_,
+and therefore Viz knows what to draw.
+(Note that surface draw quads contain surface ids only, and not textures.)
+
+### Intermediate render passes
+
+Some visual effects, such as many filters or advanced blend modes,
+require that two or more quads are drawn to an intermediate texture.
+Then the intermediate texture is drawn into a destination buffer on the GPU (or possibly another intermediate texture),
+applying the visual effect at the same time.
+To allow for this, a compositor frame actually contains a list of render passes.
+There is always a root render pass,
+which is drawn last and whose destination corresponds to the frame buffer,
+and there may be more.
+
+The possibility of multiple render passes explains the name
+"render pass"—each pass has to be executed sequentially on the GPU, in multiple "passes",
+whereas a single pass can be completed in a single massively parallel GPU computation.
+
+### Aggregation
+
+Multiple compositor frames are submitted to Viz,
+and they need to be drawn to the screen together.
+This is accomplished by an aggregation phase that converts them into a single,
+aggregated compositor frame.
+Aggregation replaces surface draw quads by the compositor frames they specify.
+It is also an opportunity to optimize away unnecessary intermediate textures or content that is offscreen.
+For example, in many cases the compositor frame for a site isolated iframe
+does not need its own intermediate texture,
+and can be drawn directly into the frame buffer via appropriate draw quads.
+The aggregation phase figures out such optimizations
+and applies them based on global knowledge not accessible to individual render compositors.
+
+### Example
+
+Here are the actual compositor frames that represent the example from the beginning of this post.
+
+```json
```

Comment:
Or just an HTML table with background colors to indicate nested render passes?

Honestly I think either that or the original nested bullets would be fine. This is a detail that only some people will read closely, and they can figure out what it means.